### ISSUE TITLE
Process finished with exit code 134 when applying random transformations for data augmentation on a dataset on apple m1 Mac

### ISSUE THREAD

**Issue by kp3393** [tensorflow/tensorflow#58840]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.9.2

### Custom Code

Yes

### OS Platform and Distribution

macOS-13.0.1-arm64-arm-64bit

### Mobile device

No

### Python version

3.10.8

### Bazel version

No

### GCC/Compiler version

No

### CUDA/cuDNN version

No

### GPU model and memory

No

### Current Behaviour?

```shell
Training of the neural network stops abruptly when applying data augmentation using tf.image method on macOS 13.0.1 with an apple m1 chip.

I installed TensorFlow for apple silicon precisely as per the instructions from the official website: 
https://developer.apple.com/metal/tensorflow-plugin/

TensorFlow-related packages and their versions:
tensorboard               2.9.1 
tensorboard-data-server   0.6.1 
tensorboard-plugin-wit    1.8.1 
tensorflow-datasets       4.7.0  
tensorflow-deps           2.9.0 
tensorflow-estimator      2.9.0 
tensorflow-macos          2.9.2 
tensorflow-metadata       1.12.0  
tensorflow-metal          0.5.1  


The code can be quickly produced on Apple m1 by just implementing the sample code provided by TF in their documentation for the Data augmentation tutorial.
Here is the link for the TF data augmentation tutorial: 
https://www.tensorflow.org/tutorials/images/data_augmentation#overview
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import tensorflow_datasets as tfds
from tensorflow.keras import layers

(train_datasets, val_ds, test_ds), metadata = tfds.load(
    'tf_flowers',
    split=['train[:80%]', 'train[80%:90%]', 'train[90%:]'],
    with_info=True,
    as_supervised=True,
)

num_classes = metadata.features['label'].num_classes
IMG_SIZE = 180
batch_size = 32
AUTOTUNE = tf.data.AUTOTUNE


def resize_and_rescale(image, label):
  image = tf.cast(image, tf.float32)
  image = tf.image.resize(image, [IMG_SIZE, IMG_SIZE])
  image = (image / 255.0)
  return image, label


def augment(image_label, seed):
  image, label = image_label
  image, label = resize_and_rescale(image, label)
  image = tf.image.resize_with_crop_or_pad(image, IMG_SIZE + 6, IMG_SIZE + 6)
  # Make a new seed.
  new_seed = tf.random.experimental.stateless_split(seed, num=1)[0, :]
  # Random crop back to the original size.
  image = tf.image.stateless_random_crop(
      image, size=[IMG_SIZE, IMG_SIZE, 3], seed=seed)
  # Random brightness.
  image = tf.image.stateless_random_brightness(
      image, max_delta=0.5, seed=new_seed)
  image = tf.clip_by_value(image, 0, 1)
  return image, label


# Create a generator.
rng = tf.random.Generator.from_seed(123, alg='philox')


# Create a wrapper function for updating seeds.
def f(x, y):
  seed = rng.make_seeds(2)[0]
  image, label = augment((x, y), seed)
  return image, label


train_ds = (
    train_datasets
    .shuffle(1000)
    .map(f, num_parallel_calls=AUTOTUNE)
    .batch(batch_size)
    .prefetch(AUTOTUNE)
)


val_ds = (
    val_ds
    .map(resize_and_rescale, num_parallel_calls=AUTOTUNE)
    .batch(batch_size)
    .prefetch(AUTOTUNE)
)

test_ds = (
    test_ds
    .map(resize_and_rescale, num_parallel_calls=AUTOTUNE)
    .batch(batch_size)
    .prefetch(AUTOTUNE)
)


model = tf.keras.Sequential([
  layers.Conv2D(16, 3, padding='same', activation='relu'),
  layers.MaxPooling2D(),
  layers.Conv2D(32, 3, padding='same', activation='relu'),
  layers.MaxPooling2D(),
  layers.Conv2D(64, 3, padding='same', activation='relu'),
  layers.MaxPooling2D(),
  layers.Flatten(),
  layers.Dense(128, activation='relu'),
  layers.Dense(num_classes)
])


model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])


epochs=5
history = model.fit(
  train_ds,
  validation_data=val_ds,
  epochs=epochs
)
```

### Relevant log output

```shell
Metal device set to: Apple M1

systemMemory: 16.00 GB
maxCacheSize: 5.33 GB

2022-12-09 22:05:22.896356: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:305] Could not identify NUMA node of platform GPU ID 0, defaulting to 0. Your kernel may not have been built with NUMA support.
2022-12-09 22:05:22.896444: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:271] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 0 MB memory) -> physical PluggableDevice (device: 0, name: METAL, pci bus id: <undefined>)
2022-12-09 22:05:23.163669: W tensorflow/core/platform/profile_utils/cpu_utils.cc:128] Failed to get CPU frequency: 0 Hz
2022-12-09 22:05:23.163761: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:113] Plugin optimizer for device_type GPU is enabled.
2022-12-09 22:05:23.175752: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:113] Plugin optimizer for device_type GPU is enabled.
Epoch 1/5
2022-12-09 22:05:23.429391: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:113] Plugin optimizer for device_type GPU is enabled.
 8/92 [=>............................] - ETA: 6s - loss: 2.2291 - accuracy: 0.1680
Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```
</details>

**Comment by mohantym** [comment]
Hi @kp3393 !
Sorry for the late response.

I could replicate this issue with 2.9 and 2.10 macos wheel and getting an accuracy of 0.7446 with 2.11 macos wheel.

I just changed the second step as below.
`python -m pip install tensorflow-macos==2.11`

Attached [screenshot](https://user-images.githubusercontent.com/86464649/207114681-0c3c03b9-84b5-4f1c-ad5b-15f0cb1a9057.png) for reference.


Thank you!

**Comment by kp3393** [comment]
Dear @mohantym,
Thank you for your prompt reply. 
I follow the same procedure; instead of the bash script, I use [Miniconda3 macOS Apple M1 64-bit pkg](https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.pkg) installer. From my understanding, it should not make a difference. 
Can you also please share your OS Platform and Distribution along with the list of packages?

Thank you!

**Comment by mohantym** [comment]
Ok @kp3393 !
Here is my os details .
```
 import platform
>>> platform.machine()
'arm64'
>>> platform.version()
'Darwin Kernel Version 21.6.0: Thu Sep 29 20:13:56 PDT 2022; root:xnu-8020.240.7~1/RELEASE_ARM64_T6000'
>>> platform.system()
'Darwin'
>>> platform.processor()
'arm'
>>> platform.platform()
'macOS-12.6.1-arm64-arm-64bit'
```

and list of packages when i enter "pip list"

```
absl-py                  1.2.0
astunparse               1.6.3
cached-property          1.5.2
cachetools               5.2.0
certifi                  2022.9.24
charset-normalizer       2.1.1
dill                     0.3.6
etils                    0.9.0
flatbuffers              22.12.6
gast                     0.4.0
google-auth              2.12.0
google-auth-oauthlib     0.4.6
google-pasta             0.2.0
googleapis-common-protos 1.57.0
grpcio                   1.49.1
h5py                     3.7.0
idna                     3.4
importlib-resources      5.10.1
keras                    2.11.0
Keras-Preprocessing      1.1.2
libclang                 14.0.6
Markdown                 3.4.1
MarkupSafe               2.1.1
numpy                    1.23.3
oauthlib                 3.2.1
opt-einsum               3.3.0
packaging                21.3
pip                      22.3.1
promise                  2.3
protobuf                 3.19.6
pyasn1                   0.4.8
pyasn1-modules           0.2.8
pyparsing                3.0.9
requests                 2.28.1
requests-oauthlib        1.3.1
rsa                      4.9
setuptools               65.5.1
six                      1.16.0
sklearn                  0.0.post1
tensorboard              2.11.0
tensorboard-data-server  0.6.1
tensorboard-plugin-wit   1.8.1
tensorflow-datasets      4.7.0
tensorflow-estimator     2.11.0
tensorflow-macos         2.11.0
tensorflow-metadata      1.12.0
termcolor                2.0.1
toml                     0.10.2
tqdm                     4.64.1
typing_extensions        4.4.0
urllib3                  1.26.12
Werkzeug                 2.2.2
wheel                    0.38.4
wrapt                    1.14.1
zipp                     3.11.0

```

Please check with TF 2.11 and metal plugin instructions as mentioned in above [comment ](https://github.com/tensorflow/tensorflow/issues/58840#issuecomment-1347970672)and let us know whether the issue still persists.

Thank you!

**Comment by kp3393** [comment]
Hello @mohantym,
I see where the problem is. You are still at macOS Monterey (your platform is **macOS-12.6.1-arm64-arm-64bit**). 
Please, refer to my problem description. I have mentioned my platform to be **macOS-13.0.1-arm64-arm-64bit**. If you update to macOS Ventura, there is a high possibility that you will face the same issues.

**Comment by kp3393** [comment]
Hello @mohantym @sachinprasadhs,
Any updates? I will be grateful for some help to solve this issue.

Thank you for your time!

**Comment by kp3393** [comment]
Commenting on the thread to prevent it from closing as stale. Please, will be grateful for any updates.

**Comment by sachinprasadhs** [comment]
Apologies for the delayed response. 
It looks like the memory issue on your system side. 
Could you please try with less number of epochs or try the same example mentioned in the document here https://www.tensorflow.org/tutorials/images/data_augmentation#overviewwith 5 epochs and let us know if you still face the error. Thanks!
